python/pyiceberg/table/snapshots.py

```diff
@@ -117,12 +125,369 @@ def manifests(self, io: FileIO) -> List[ManifestFile]:
             return list(read_manifest_list(file))
         return []
 
+    def added_data_files(self, io: FileIO) -> Generator[DataFile, None, None]:
+        for manifest in self.manifests(io):
+            yield from [entry.data_file for entry in manifest.fetch_manifest_entry(io)]
+
 
 class MetadataLogEntry(IcebergBaseModel):
     metadata_file: str = Field(alias="metadata-file")
     timestamp_ms: int = Field(alias="timestamp-ms")
 
 
 class SnapshotLogEntry(IcebergBaseModel):
-    snapshot_id: str = Field(alias="snapshot-id")
+    snapshot_id: int = Field(alias="snapshot-id")
     timestamp_ms: int = Field(alias="timestamp-ms")
+
+
+def is_ancestor_of(table: "Table", snapshot_id: int, ancestor_snapshot_id: int) -> bool:
```

Comment:
This is to avoid circular dependencies. We only import Table when doing type checking:
```python
if TYPE_CHECKING:
    from pyiceberg.table import Table
```
The Table has snapshots, and the snapshot refers back to the table. By putting this in quotes, we still get type-checking, but it doesn't introduce a circular dependency.